patches/server/0521-Cache-block-data-strings.patch

```diff
@@ -17,23 +17,29 @@ index 4af56ca699d456eb23794673861323ab06cf6de8..bb18848f2b91b80ab5099c00b423738b
  
          if (this.isSameThread()) {
 diff --git a/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java b/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
-index 3594f432a25b580173e8577bf324be954f5eddd1..6dc8f9f269db6971b8b46819e017357899ccd118 100644
+index 3594f432a25b580173e8577bf324be954f5eddd1..dc51181dde6e49d2cbed9f21c007b8256a8910c6 100644
 --- a/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
 +++ b/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
-@@ -13,6 +13,7 @@ import net.minecraft.commands.arguments.blocks.BlockStateParser;
+@@ -8,11 +8,13 @@ import com.mojang.brigadier.exceptions.CommandSyntaxException;
+ import java.util.HashMap;
+ import java.util.Map;
+ import java.util.Set;
++import java.util.concurrent.ConcurrentHashMap;
```

Comment:
While the patch suggests otherwise, this should be a fully qualified import instead.
Maybe while fixing this patch, you could just remove the other import as well to clean up the patch.

A simple 
```patch
- ResourceLocation key = Registry.BLOCK.getKey(block);
+ var key = Registry.BLOCK.getKey(block);
``` 
would suffice already to also removed the resource location import.

Reply:
Good point, makes merging with upstream easier